### DISCUSSION TITLE
Are delayed messages ready for high throughput?

### DISCUSSION THREAD

**Discussion by dszymon** [cloudamqp/lavinmq#782] [Q&A] [answered]
Do they support persistence and offer the same performance as normal messages?

**Comment by viktorerlingsson** [comment, marked as answer]
Hi, 
Yes, delayed message exchanges support persistence. 
Performance will be slightly lower than with other types of exchanges, because there is some additional processing that needs to be done. Delayed messages are first published to a separate queue, and then re-published (dead-lettered) to the correct queue with the supplied routing key. 

Hope that answers your question!

**Reply by dszymon** [reply]
thanks for the answer
